File: android/app/src/main/java/com/coinswallet/coins/CosmosCoin.java
// EthereumCoin.java
package com.coinswallet.coins;

import com.coinswallet.CoinFactory;
import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.ReadableArray;
import com.facebook.react.bridge.WritableArray;
import com.facebook.react.bridge.WritableMap;

import wallet.core.jni.CoinType;
import wallet.core.jni.PrivateKey;

public class CosmosCoin extends CoinFactory.Coin {
    private final String address;
    private final PrivateKey privateKey;

    public CosmosCoin(String mnemonic) {
        super(mnemonic);
        this.address = CoinType.COSMOS.deriveAddress(wallet.getKeyForCoin(CoinType.COSMOS));
        this.privateKey = wallet.getKeyForCoin(CoinType.COSMOS);
    }

    @Override
    public String getNewAddress(Boolean isTestNet) {
        return this.address;
    }

    @Override
    public String getPrivateKey(Boolean isTestNet) {
        return CoinFactory.Coin.toHex(this.privateKey.data());
    }
    public String signTransaction(String rawData) {
        return null;
    }
}

File: ios/CosmosCoin.swift
//
//  EthereumCoin.swift
//  coinswallet
//
//  Created by Tal Grynbaum on 30/06/2023.
//

import Foundation
import WalletCore

class CosmosCoin: CoinFactory.Coin {
  private var address: String = ""
  private var privateKey: String = ""
  
  override init(mnemonic: String) {
    super.init(mnemonic: mnemonic)
    let privateKeyInfo = wallet.getKeyForCoin(coin: .cosmos)
    self.address = CoinType.cosmos.deriveAddress(privateKey: privateKeyInfo)
    self.privateKey = privateKeyInfo.data.hexString;
  }
  
  override func getNewAddress(isTestNet:Bool) -> String {
    return self.address
  }
  
  override func getPrivateKey(isTestNet:Bool) -> String {
    return self.privateKey;
  }
  
  override func signTransaction(rawData: String) -> String {
    // Implement the logic to sign an Ethereum transaction using the provided raw data
    return ""
  }
}
